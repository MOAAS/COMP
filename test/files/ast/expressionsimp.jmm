import static io.println(int);
import static io.println(boolean);

class ExpressionSimp {
	public boolean veryComplexCalculationBool() {
	    return true;
	}

    public int veryComplexCalculationInt() {
        return 0;
    }

	public static void main(String[] args) {
	    ExpressionSimp obj;
	    int a;
	    boolean b;
	    int unknownInt;
	    boolean unknownBool;

	    obj = new ExpressionSimp();
        unknownInt = obj.veryComplexCalculationInt();
        unknownBool = obj.veryComplexCalculationBool();
	    a = 1 + 2 + 3; // 6
	    a = 455 + 20 * (10 * 20 - 42 / 3) - 1;
	    a = unknownInt * 0 * unknownInt + unknownInt * 0; // 0
	    a = unknownInt + 0 - (0 / unknownInt); // unknownInt
	    a = unknownInt * 1; // unknownInt
	    a = unknownInt / unknownInt; // 1

        b = true && unknownBool; // unknownBool
        b = false && unknownBool; // false
        b = true && true && true && true && true && true; // true
        b = true && true && true && true && true && false; // false
        b = 1 < 2 && !(5 * 2 + 1 < 10); // true


        // estes ficam
        a = 0 * obj.veryComplexCalculationInt();
        b = obj.veryComplexCalculationBool() && false;
	}
}
