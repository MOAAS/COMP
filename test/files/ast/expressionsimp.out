Program
 ImportDeclaration
  ImportLifetime
  ImportName
   io
   println
  ImportArgs
   int
  ImportType
 ImportDeclaration
  ImportLifetime
  ImportName
   io
   println
  ImportArgs
   boolean
  ImportType
 ClassDeclaration
  ClassInfo
   ExpressionSimp
  RegularMethodDeclaration
   MethodType
    boolean
   MethodName
    veryComplexCalculationBool
   MethodVars
   Return
    True
  RegularMethodDeclaration
   MethodType
    int
   MethodName
    veryComplexCalculationInt
   MethodVars
   Return
    0
  MainDeclaration
   ArgName
    args
   MethodContent
    VarDeclaration
     ExpressionSimp
     obj
    VarDeclaration
     int
     a
    VarDeclaration
     boolean
     b
    VarDeclaration
     int
     unknownInt
    VarDeclaration
     boolean
     unknownBool
    Assignment
     obj
     ObjectInit
      ExpressionSimp
      ObjectInitArgs
    Assignment
     unknownInt
     Dot
      obj
      DotArgs
       veryComplexCalculationInt
    Assignment
     unknownBool
     Dot
      obj
      DotArgs
       veryComplexCalculationBool
    Assignment
     a
     6
    Assignment
     a
     4174
    Assignment
     a
     0
    Assignment
     a
     unknownInt
    Assignment
     a
     unknownInt
    Assignment
     a
     1
    Assignment
     b
     unknownBool
    Assignment
     b
     False
    Assignment
     b
     True
    Assignment
     b
     False
    Assignment
     b
     True
    Assignment
     a
     Mul
      0
      Dot
       obj
       DotArgs
        veryComplexCalculationInt
    Assignment
     b
     And
      Dot
       obj
       DotArgs
        veryComplexCalculationBool
      False
